### PR TITLE
Consistent braces requirements for hash parameters

### PR DESCRIPTION
**Pull request by adsteel**
[Docs for the cop are here](https://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/Cop/Style/BracesAroundHashParameters). My argument is that braces make it very clear what the argument is, and don't leave interpretation to the magic of the MRI. This also simplifies the cognitive overhead of coding, where you don't have to stop to think "should this hash argument have braces or not?". It always should.

As always, if folks have differing opinions we could also stop enforcing this entirely.

tldr: always require braces around the final argument of a method if it is a hash. like this:

```ruby
some_method(x, y, {a: 1, b: 2})
```
Not like this:

```ruby
some_method(x, y, a: 1, b: 2)
```

### CODE REVIEW

**Review by derekclee — APPROVED** [review]
If this only applies to the last arg, I'm good with this change. It there is only 1 arg,  I think the shorthand is fine.

**Comment by mhuggins** [comment]
I would rather we not enforce this than to enforce braces.  I have a conflicting view that it takes longer to parse with the braces.  However, in scenarios where two hashes are passed, I prefer explicit brackets, e.g.:

```ruby
# better:
index { field: 'foo' }, { background: true }

# worse:
index { field: 'foo' }, background: true
```

**Comment by pk-nb** [comment]
I'm low on this, but I feel this rule would be better to let the dev decide what's more readable 
 case by case rather than enforce.

**Comment by adsteel** [comment]
Sounds like a majority. I'll update to a non-enforcement.

**Comment by adsteel** [comment]
Updated to non-enforcement and ready for re-review @pk-nb @mhuggins and @derekclee.

**Review by pk-nb — APPROVED** [review]
I'm good with it, but will let people that spend more time in ruby give the final say

**Review by mhuggins — APPROVED** [review]

**Comment by derekclee** [comment]
👍

**Review by ianhenrysmith — APPROVED** [review]
I'm good with this. Leaving braces up to the dev allows for more expressiveness here.
